### ISSUE TITLE
Feature for warnings

### ISSUE THREAD

**Issue by ramack** [martykan/forecastie#192]
It would be great, if forcastie would warn me if a configured event is upcoming. Could be "frost" - right now summer is coming but in autumn it would help me to bring my flowers into the warm garage. For sure it is not helpful to be warned the whole winter time, so a setting would be good to enable the warning only after e. g. 20 day without frost.

Or on a hike ore bicycle tour it would be good to be warned of upcoming rain, telling me "rain in 35 min" - I don't know whether this is achievable with the current OWM data, but if there is a whether station in the direction from where the wind comes it should be possible to give such warnings.

And maybe also show official warnings.

**Comment by ildar** [comment]
Yandex weather warns about any weather change that may be important for a
man: rapid warm/cold, rain etc. No need to configure anything.

**Comment by ramack** [comment]
maybe, but this seems not to be free software, isn't it?

**Comment by ildar** [comment]
Sure it's not. Just an example of UX

**Comment by FridoDeluxe** [comment]
OWM has a new beta feature called Triggers or Weather Alerts: http://openweathermap.org/triggers.
They don't provide predefined alerts, instead the user can define certain conditions when to be alerted by polling the API. I'm not sure if the possibilities of this trigger API are sufficient for what you have in mind for weather warnings?

**Comment by ramack** [comment]
Yes, this seems to be exactly what I thought about. Even though a easy but still powerful UI for setting these is probably not straight forward.

**Comment by FridoDeluxe** [comment]
For the beginning a few built-in general triggers (extreme high/low temperature, heavy rain, thunderstorm...) inside Forecastie should be enough, they could be selected by the user if desired.
But I see potential problems with the API key: If at a later stage triggers can be user-defined they are saved into the OWM account of the API key, additionally keeping track of weather triggers means probably more frequent polling and thus more API calls. This will not work with the built-in API key, if the users don't enter their own key (see #182).

**Comment by ildar** [comment]
Then this feature might require the user API key

**Comment by robinpaulson** [comment]
@FridoDeluxe raises a good point; since we got an upgraded API key it is likely not an issue. We can try, see what happens and then if there are problems, only allow the warning feature if the user has their own key.

**Comment by robinpaulson** [comment]
Thinking out loud, I would suggest the two pieces of information required for an alert (other than location, which for the moment I will assume is fixed) are the weather condition the user wishes to be alerted of and the time frame. For a first cut, I suggest the time frame should be limited to "During the next <n> hours/days", where <n> is configurable. Perhaps we make that more sophisticated later on. The weather condition we can break down further into two things: a particular quality (temperature, rain amount, cloud cover, wind speed) and a limit with greater than/less than operator. So for example: "Alert me if there is more than 5mm rain per 3hr block, in the next 12 hours"; "Alert me if the temperature will be less than 2 degrees Centigrade in the next 5 days". The initial point about deviations from the norm by @ramack is useful, perhaps we add that in later.

**Comment by ildar** [comment]
I'd be happy with 7 a.m. notification if a rain or snow or heavy wind is
expected during the day.
The notification is useless after 8 a.m. then I'm outside.
Simple and useful.

**Comment by Currrupted** [comment]
> 
> I'd be happy with 7 a.m. notification if a rain or snow or heavy wind is expected during the day. The notification is useless after 8 a.m. then I'm outside. Simple and useful.

I love your ideas so far! How do the alerts by OWM look like exactly? Because the only implementation I have seen so far is in the official weather application of the federal weather service in Germany:
![grafik](https://user-images.githubusercontent.com/13292623/70044976-9c1a5980-15c3-11ea-83a4-3254062c34aa.png)

**Comment by Lee-Carre** [comment]
See also thuryn/your-local-weather#109 in which I posted a bunch of thoughts & ideas on this concept.

**Comment by ildar** [comment]
check this: https://f-droid.org/en/packages/org.breezyweather/

**Comment by ildar** [comment]
not an issue.
These apps are not for emergency situations. It's so ... unsuitable .. to
trust your **life** to a general purpose weather application.

**Comment by ildar** [comment]
On Thu, Jul 25, 2024 at 4:19 PM jayb-g ***@***.***> wrote:

> I think there should be weather alerts included in the app, even be it
> without push. Then its upto users how frequently they want it to refresh as
> usual.
>
Then it's BreezyWeather you need.
